src/memory.rs

```diff
@@ -0,0 +1,60 @@
+use std::cell::RefCell;
+
+use rustc_mir::interpret::{Pointer, Allocation, AllocationExtra, InterpResult};
+use rustc_target::abi::Size;
+
+use crate::{stacked_borrows, intptrcast};
+use crate::stacked_borrows::{Tag, AccessKind};
+
+#[derive(Default, Clone, Debug)]
+pub struct MemoryState {
+    pub stacked: stacked_borrows::MemoryState,
+    pub intptrcast: intptrcast::MemoryState,
+    pub seed: Option<u64>,
```

Comment:
Move the `rng` field from https://github.com/rust-lang/miri/blob/c65fbc49d7b9269b277b7207f1044ee851f16b16/src/lib.rs#L360 to here (really remove it from the other site) instead of storing the `seed`. Any code that has access to the `Evaluator` has access to `Memory`, so it should all work out by bending the `ecx.extra.rng` invocations to `ecx.memory.extra.rng` invoations